README.md

```diff
@@ -1,22 +1,134 @@
+     ```
+
+1. **Configure Environment Variables**:
+   - Open the `backend/.env/local` file in your preferred text editor and change the `DJANGO_CONFIGURATION` value to `Local`:
+     ```plaintext
+     DJANGO_CONFIGURATION=Local
+     ```
+
+1. **Set Up Algolia**:
+   - Go to [Algolia](https://www.algolia.com/) and create a free account.
+   - After creating an account, create an Algolia app.
+   - Update your `.env/local` file with the following keys from your Algolia app:
+     ```plaintext
+     DJANGO_ALGOLIA_API_KEY=<your_algolia_api_key>
+     DJANGO_ALGOLIA_APPLICATION_ID=<your_algolia_application_id>
+     ```
+   - Ensure that your API key has index write permissions. You can ignore any onboarding wizard instructions provided by Algolia.
+
+1. **Run the Application**:
+   - In your terminal, navigate to the project directory and run the following command to start the application:
+     ```bash
+     make run
+     ```
+   - Leave this terminal session running and wait until you see that [Nest local](http://localhost:8000/api/v1) is responding.
+
+1. **Load Initial Data**:
+   - Open a new terminal session and run the following command to populate the database with initial data from fixtures:
+     ```bash
+     make load-data
+     ```
+
+1. **Index Data**:
+   - In the same terminal session, run the following command to index the data:
+     ```bash
+     make index-data
+     ```
+
+1. **Verify API Endpoints**:
+   - Check that the data is available via these API endpoints:
+     - [Projects Endpoint](http://localhost:8000/api/v1/owasp/search/project)
+     - [Issues Endpoint](http://localhost:8000/api/v1/owasp/search/issue)
+
+### Optional Steps (for managing or fetching data):
+
+If you plan to manage or fetch data, follow these additional steps:
+
+1. **Create a Super User**:
+   - Run the following command to create a super user for accessing the admin interface:
+     ```bash
+     make setup
+     ```
+
+1. **Generate a GitHub Personal Access Token**:
+   - Create a GitHub [personal access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens).
+
+1. **Update Environment Variables with GitHub Token**:
+   - Open `backend/.env/local` again and update it with your GitHub token:
+     ```plaintext
+     GITHUB_TOKEN=<your_github_token>
+     ```
+
+1. **Sync Local Database Data**:
+   - Now you should be able to run the following command to sync your local database data with GitHub:
+     ```bash
+     make sync
+     ```
+
+
+## Linting and Formatting Setup
+
+To ensure code quality, we utilize ESLint and Prettier with Husky for pre-commit hooks.
+
+### Installing Dependencies
+
+Make sure you have the necessary dependencies installed:
+
+```bash
+npm install --save-dev eslint prettier husky lint-staged eslint-plugin-react eslint-plugin-react-hooks @typescript-eslint/eslint-plugin @typescript-eslint/parser eslint-config-prettier eslint-plugin-prettier
```

Comment:
This should be installed automatically upon docker stack run, right?
if `npm install` takes care of this we can remove it from here.

Reply:
When you would update the `package.json`  then these dependencies will automatically get installed right?
So I feel and you would see the same that `npm install` must install all these dependencies rather than explicitly doing it.

Reply:
> umm, my thought process behind putting these commands in readme was to provide users with a short section for commands and required dependencies, as if someone had problems setting up the project, they could simply follow the mentioned instructions. Please do suggest what feels best here

This would suit better for a troubleshooting section (when we have one). For the initial setup the fewer instructions the better. Ideally one command per a setup step.

Reply:
@harsh3dev @arkid15r yes, after `npm install` all the dependencies will be automatically installed. On running `npm run prepare` husky will be installed and `.husky/pre-commit` file will be created with the required content already present inside. 
We'll only have to run a single husky command to execute linting aswell as prettier formatting.
I mentioned the ESLint and Prettier commands in the `README.md` so that if someone wanted to test run either of them, they could easily do it without having to look for the right command.